src/javascript/Edit/startWorkflow/startWorkflow.action.jsx

```diff
@@ -24,8 +14,20 @@ export default composeActions(
     reduxAction(mapStateToContext),
     {
         init: context => {
-            context.enabled = shouldEnableAction(context.isMainButton, context);
-            context.disabled = context.nodeData.lockInfo.isLocked || context.formik.dirty;
+            if (context.isMainButton) {
+                context.enabled = context.mode === Constants.routes.baseEditRoute &&
+                    !context.nodeData.hasPublishPermission &&
+                    context.nodeData.hasStartPublicationWorkflowPermission;
+                context.disabled = context.nodeData.lockInfo.isLocked || context.formik.dirty;
+            } else {
+                context.enabled = context.mode === Constants.routes.baseEditRoute && context.nodeData.hasPublishPermission;
+
+                // In case the action is in the menu, formik is in parent context
+                if (context.enabled && (context.nodeData.lockInfo.isLocked || context.parent.formik.dirty)) {
```

Comment:
I thinks there is some test missing.

This code with a lot of "if" is hard to understand, test should be easier for me to understand this.